docs/workforce-human-oidc-auth.md

```diff
@@ -0,0 +1,172 @@
+
+Under normal circumstances, the application provides an access token received from the Identity Provider to the driver. The application SHOULD provide a toggle to users that can be used to indicate that it should use the ID token, rather than the access token, in its place.[^2]
+
+## OIDC Scopes
+
+The initial request to the IdP involves specifying the `scope` parameter. This is a space-separated unordered list of strings which contains:
+
+* The scopes listed in the `requestScopes` field of the IdP response from the MongoDB endpoint, and
+* The `openid` and `offline_access` scopes[^3]. If the IdP metadata document contains a `scopes_supported` field, and this field does not list one of these scopes, then the respective scope SHOULD NOT be added to the `scope` parameter.
+
+# Authorization Code Flow
+
+In this flow, the client application uses a local HTTP server to receive a response from the Identity Provider, as described in [RFC8252](https://datatracker.ietf.org/doc/html/rfc8252).[^4] The application follows these steps:
+
+1. Generate a code challenge for PKCE using cryptographically random data, as described in [RFC7636](https://datatracker.ietf.org/doc/html/rfc7636).
+2. Spin up a local HTTP server. The default (incoming) redirect URL for MongoDB applications is `http://localhost:27097/redirect`, which MAY be configurable. If the application allows configuring the URL, the port MAY be specified as `0` to allow listening on an arbitrary port. The application listens on the host and port listed in the URL. The application MUST listen on all addresses that the hostname resolves to through `getaddrinfo()`, and MUST listen on the same port in all cases. If listening on any address fails, or `getaddrinfo()` did not return any addresses, it must abort this process and the application MAY fall back to Device Auth.
+   1. To preserve privacy, pages served from this local HTTP server MUST NOT reference external resources directly. They MAY provide external links that require user interaction.
+   2. The HTTP server SHOULD set default headers of `Content-Security-Policy: default-src 'self'` and `Referrer-Policy: no-referrer`, or similarly restrictive defaults.
+3. Compose the URL for Auth Code Flow, using the code challenge generated earlier and the redirect URL from the previous step, with the port replaced with the actual port if it was specified as `0`.
+   1. The application MUST add a `state` parameter containing cryptographically random data ([RFC6749](https://datatracker.ietf.org/doc/html/rfc6749#section-4.1.1) specifies this as optional but recommended).
+   2. The application MAY allow the user to indicate that it should add a `nonce` parameter containing cryptographically random data to the authentication request, as defined in the OpenID connect specification, which is then later embedded in the ID token itself.
```

Comment:
This item is currently under active investigation, [MONGOSH-1905](https://jira.mongodb.org/browse/MONGOSH-1905) is essentially a user request for this, on the other hand we've had tickets like [MONGOSH-1669](https://jira.mongodb.org/browse/MONGOSH-1669) where we've been taking a stance where we don't strictly require OIDC features and are happy with OAuth-2.0-only features. 

The text of the OIDC spec ("If present in the ID Token, Clients MUST verify that the nonce Claim Value is equal to the value of the nonce parameter sent in the Authentication Request.") does imply to me that the IdP cannot, on its own, add a `nonce` value that the client did not specify to begin with and still be considered spec-compliant. 

Now ... _maybe_ adding `nonce`s is something that all relevant identity providers support anyway, in which case we're fine to turn this line into a SHOULD/MUST and remove the user interaction, but we'd likely need to verify this first. So that's why, for now, this is the wording I'm going with here.